com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -872,12 +872,21 @@ private void downloadSymbols(IProgress progress) throws IOException, CompileExce
         }
     }
 
+    private void registerPipelinePlugins() {
+        // Find the plugins and register them now, before we're building the content
+        List<File> plugins = ExtenderUtil.getPipelinePlugins(this);
+        for (File plugin : plugins) {
+            Bob.getClassLoaderScanner().addUrl(plugin);
+            logInfo("Using plugin %s", plugin);
+        }
+        Bob.rescanClasses(this);
+    }
+
     private List<TaskResult> doBuild(IProgress monitor, String... commands) throws IOException, CompileExceptionError, MultipleCompileException {
         fileSystem.loadCache();
         IResource stateResource = fileSystem.get(FilenameUtils.concat(buildDirectory, "state"));
         state = State.load(stateResource);
-        createTasks();
-        validateBuildResourceMapping();
+
```

Comment:
We cannot instansiate the builders _before_ we've (potentially) built them on the build server!